src/HttpRequest.cpp

```diff
@@ -498,6 +498,32 @@ stringpair_t	HttpRequest::parseHeader(octets_t header)
 	return (authority);
 }
 
+void	HttpRequest::validateContentType(const Location &location)
+{
+	stringmap_t::iterator											contentType;
+	std::string														contentTypeValue;
+	std::map< std::string, std::set<std::string> >					mimeTypesDict;
+	struct stat														statBuff;
+
+	contentType = this->headerFields.find("content-type");
+	mimeTypesDict = location.getMimeTypes().getMimeTypesDict();
+	if (stat(this->targetResource.c_str(), &statBuff) != 0)
+		throw (ResponseException(500, "Internal Server Error"));
```

Comment:
ResponseException should have something like "Failure on stat()" as the second argument, in order to provide more context